src/Lit/Experimental.fs

```diff
@@ -0,0 +1,76 @@
+[<Experimental("This is an exploration space, anything can be added or removed in this module at any point")>]
+module Lit.Experimental
+
+
+open Fable.Core
+open Fable.Core.JsInterop
+open Lit
+
+[<AttachMembers>]
+type StateController<'T>(host, initial: 'T) =
+    inherit ReactiveController(host)
+    let mutable state = initial
+
+    member _.Value = state
+
+    member _.SetState(value: 'T) =
+        state <- value
+        host.requestUpdate ()
```

Comment:
I'm playing with the idea of having hook-like abstractions but rather than backed by a directive backed by ReactiveControllers they would work nicely for both class function based LitElements, too crazy perhaps?